blueocean-rest/src/main/java/io/jenkins/blueocean/analytics/AdditionalAnalyticsProperties.java

```diff
@@ -0,0 +1,18 @@
+package io.jenkins.blueocean.analytics;
+
+import hudson.ExtensionPoint;
+import io.jenkins.blueocean.analytics.Analytics.TrackRequest;
+
+import java.util.Map;
+
+/**
+ * Enhance a request with properties when tracking an analytics event
+ */
+public abstract class AdditionalAnalyticsProperties implements ExtensionPoint {
+    /**
+     * Add additional properties for a {@link TrackRequest}
+     * @param trackReq to be tracked
+     * @return additional properties to be sent with the request or null
+     */
+    public abstract Map<String, Object> properties(TrackRequest trackReq);
```

Comment:
@i386 minor nit pick, TrackRequest should be in it's own class vs currently its static inner.